src/core/deployer.sol

```diff
@@ -68,93 +62,101 @@ contract LightSwitchFab {
     }
 }
 
-contract PileFab {
-   function newPile(address tkn, address title, address debtRegister) public returns (Pile pile) {
-        pile = new Pile(tkn, title, debtRegister);
-        pile.rely(msg.sender);
-        pile.deny(address(this));
-    }
-}
-
 contract ShelfFab {
-   function newShelf(address pile, address appraiser, address title) public returns (Shelf shelf) {
-        shelf = new Shelf(pile, appraiser, title);
+   function newShelf(address tkn_, address title_, address debt_, address Principal_) public returns (Shelf shelf) {
+        shelf = new Shelf(tkn_, title_, debt_, Principal_);
         shelf.rely(msg.sender);
         shelf.deny(address(this));
     }
 }
 
 contract DeskFab {
     // note: this is the mock Desk, which will interface with the lender/tranche side of Tinlake, and does not require auth for now.
```

Comment:
this note is weird, is it still relevant?

Reply:
we will remove this it anyway when integrating the new borrower & lender contracts